src/ORM/Queries/SQLConditionalExpression.php

```diff
@@ -78,7 +78,18 @@ public function addFrom($from)
         if (is_array($from)) {
             $this->from = array_merge($this->from, $from);
         } elseif (!empty($from)) {
-            $this->from[str_replace(['"','`'], '', $from)] = $from;
+            // Make sure we don't have any pointless blank space in our From clause
+            $from = trim($from);
+
+            // Check if the from clause looks like a regular table name
+            // Table name can contain only alphanumeric characters and underscores
+            if (preg_match('/^["`]?([a-zA-Z0-9_]+)["`]?$/', $from)) {
+                // Add an alias for the table name, stripping any quotes
+                $this->from[str_replace(['"','`'], '', $from)] = $from;
+            } else {
+                // Add from clause without an alias
```

Comment:
```suggestion
                // Add from clause without an alias - this is probably a full sub-select with its own explicit alias.
```
This seems to be the reasoning - lets leave it here for future generations.

Reply:
Done.